src/Fossa/API/Types.hs

```diff
@@ -26,12 +27,13 @@ import Data.Text.Encoding (encodeUtf8)
 import Network.HTTP.Req
 import Text.URI (URI, render)
 import qualified Unsafe.Coerce as Unsafe
+import Text.URI.QQ (uri)
 
 newtype ApiKey = ApiKey {unApiKey :: Text}
   deriving (Eq, Ord, Show)
 
 data ApiOpts = ApiOpts
-  { apiOptsUri :: URI,
+  { apiOptsUri :: Maybe URI,
```

Comment:
I don't fully understand why it is wrong to make this type a `Maybe`. If the value is `Nothing` we use the default endpoint value. We will always have a place to send the API requests because we have to check if the value is `Just`. The biggest difference is where we are now handling the default endpoint.

Reply:
If there's a technical challenge forcing this to be a `Maybe`, then that's fine, we can just leave a todo/fixme and work it out later.

But to my original point:  This type represents a definite intention to use the FOSSA API.  It should be present if, and only if, we have all the information we need to use the API.  That includes a URI, which may or may not be the default (this type doesn't care if it is the default, that's the job of configuration).  This way, we never have to worry about resolving the `Maybe` in functions which consume this type, and eventually need some URI.  The resolution has already happened by the time we created some value of this type.

Reply:
I think your point makes a lot of sense in the context of having a `SpectrometerConfig` type, but at the moment the way this value is actually set I don't believe leaving this as `URI` makes sense. The challenge right now is where the default value is defined, which I would be okay with defining in the `mergeFileCmdConfig` function, but that feels like a weird place to set a default value.

Reply:
Like I said, "technical challenge" is an acceptable reason to leave this as-is.  Once we get customer feedback on our configuration approach, we can revisit this.

Reply:
Honestly, I think a follow-up refactoring PR would be a good time to address this, the shortcomings are kinda clear to me after working on this.

Reply:
Also, it allows for rock-solid config testing.